Tools/i18n/pygettext.py

```diff
@@ -459,37 +475,46 @@ def _extract_docstring(self, node):
 
     def _extract_message(self, node):
         func_name = self._get_func_name(node)
-        spec = self.options.keywords.get(func_name)
-        if spec is None:
-            return
+        specs = self.options.keywords.get(func_name, [])
+        for spec in specs:
+            extracted = self._extract_message_with_spec(node, spec)
+            if extracted:
+                break
+
+    def _extract_message_with_spec(self, node, spec):
+        """Extract a gettext call with the given spec.
 
-        max_index = max(spec)
+        Return True if the gettext call was successfully extracted, False
+        otherwise.
+        """
+        max_index = max(spec.values())
         has_var_positional = any(isinstance(arg, ast.Starred) for
                                  arg in node.args[:max_index+1])
         if has_var_positional:
             print(f'*** {self.filename}:{node.lineno}: Variable positional '
                   f'arguments are not allowed in gettext calls', file=sys.stderr)
-            return
+            return False
 
         if max_index >= len(node.args):
             print(f'*** {self.filename}:{node.lineno}: Expected at least '
-                  f'{max(spec) + 1} positional argument(s) in gettext call, '
+                  f'{max_index + 1} positional argument(s) in gettext call, '
                   f'got {len(node.args)}', file=sys.stderr)
-            return
+            return False
```

Comment:
And this message will be incorrectly printed in case of multiple specs, some of them support the specified number of args, and others don't.